packages/teleport/src/Audit/EventList/EventList.tsx

```diff
@@ -14,147 +14,101 @@ See the License for the specific language governing permissions and
-    if (sortDir === Table.SortTypes.ASC) {
-      return sorted.reverse();
-    }
-
-    return sorted;
-  }, [state, events, search]);
-
-  // paginate
-  const tableProps = {
-    pageSize,
-    data,
-    fetchMore,
-    fetchStatus,
-    search,
-    onSearchChange,
-  };
-  const { detailsToShow, colSortDirs } = state;
   return (
-    <React.Fragment>
-      <PagedTable {...tableProps}>
-        <Table.Column
-          columnKey="codeDesc"
-          cell={<TypeCell clusterId={clusterId} />}
-          header={
-            <Table.SortHeaderCell
-              sortDir={colSortDirs.codeDesc}
-              onSortChange={onSortChange}
-              title="Type"
-            />
-          }
-        />
-        <Table.Column
-          columnKey="message"
-          header={<Table.Cell>Description</Table.Cell>}
-          cell={<DescCell />}
-        />
-        <Table.Column
-          columnKey="time"
-          header={
-            <Table.SortHeaderCell
-              sortDir={colSortDirs.time}
-              onSortChange={onSortChange}
-              title="Created"
-            />
-          }
-          cell={<TimeCell />}
-        />
-        <Table.Column
-          header={<Table.Cell />}
-          cell={
-            <ActionCell clusterId={clusterId} onViewDetails={showDetails} />
-          }
-        />
-      </PagedTable>
+    <>
+      <Table
+        data={events}
+        columns={[
+          {
+            key: 'codeDesc',
+            headerText: 'Type',
+            isSortable: true,
+            render: event => <TypeCell event={event} clusterId={clusterId} />,
+          },
+          {
+            key: 'message',
+            headerText: 'Description',
+            render: ({ message }) => <DescCell message={message} />,
+          },
+          {
+            key: 'time',
+            headerText: 'Created',
+            isSortable: true,
+            render: ({ time }) => <TimeCell time={time} />,
+          },
+          {
+            altKey: 'show-details-btn',
+            render: event => (
+              <ActionCell event={event} onShowDetails={setDetailsToShow} />
+            ),
+          },
+        ]}
+        emptyText={'No Events Found'}
+        isSearchable
+        initialSort={{ key: 'time', dir: 'DESC' }}
+        pagination={{ pageSize }}
+        fetching={{
+          onFetchMore: fetchMore,
+          fetchStatus,
+        }}
+      />
       {detailsToShow && (
-        <EventDialog event={detailsToShow} onClose={closeDetails} />
+        <EventDialog
+          event={detailsToShow}
+          onClose={() => setDetailsToShow(null)}
+        />
       )}
-    </React.Fragment>
+    </>
   );
 }
 
-type EventListState = {
-  searchableProps: string[];
-  colSortDirs: Record<string, string>;
-  detailsToShow?: Event;
-};
+export const ActionCell = ({
+  event,
+  onShowDetails,
+}: {
+  event: Event;
+  onShowDetails: React.Dispatch<React.SetStateAction<Event>>;
```

Comment:
Why typing with dispatcher when we can do this?

`React.Dispatch<React.SetStateAction<Event>>` -> `onShowDetails: (e: Event) => void` ?

Reply:
The type with dispatcher is technically the more accurate type, but it doesn't make much difference